File: 3918-check-divisibility-by-digit-sum-and-product/check-divisibility-by-digit-sum-and-product.cpp
class Solution {
public:
    bool checkDivisibility(int n) {
        int original_No=n;
        int sum=0,product=1;
        while(n!=0){
            int rem=n%10;
            sum+=rem;
            product*=rem;
            n/=10;
        }
        return (original_No%(sum+product)==0)?true:false;
    }
};

File: 3918-check-divisibility-by-digit-sum-and-product/check-divisibility-by-digit-sum-and-product.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        originalNo=n
        sum=0
        prod=1
        while n!=0:
            rem=n%10
            sum+=rem
            prod*=rem
            n//=10
        print(sum+prod)
        return True if originalNo%(sum+prod)==0 else False